File: educational-system-consumer-teacher-9002/src/main/java/edu/hubu/consumerteacher/service/ITeacherSelectCourseService.java
package edu.hubu.consumerteacher.service;

import edu.hubu.commons.model.CommonResult;
import edu.hubu.commons.model.teacher.TeacherSelectCourse;
import io.swagger.annotations.ApiImplicitParam;
import io.swagger.annotations.ApiImplicitParams;
import io.swagger.annotations.ApiOperation;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.*;

/**
 * @author moonlan
 * date 2021/2/1 下午4:54
 */
@SuppressWarnings(value = "unused")
@FeignClient(name = "teacher-provider-service")
public interface ITeacherSelectCourseService {

    @GetMapping("/teacherSelectCourse/{id}")
    CommonResult selectOne(@PathVariable("id") String id);

    @GetMapping("/teacherSelectCourse/all")
    CommonResult selectAll(@RequestParam(value = "orderBy", defaultValue = "teacherSelectCourseId") String orderBy);

    @GetMapping("/teacherSelectCourse/")
    CommonResult selectAllPage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "teacherSelectCourseId") String orderBy
    );

    @PostMapping("/teacherSelectCourse/select-like")
    CommonResult selectLikePage(
            @RequestParam(value = "currentPage", defaultValue = "1") Integer currentPage,
            @RequestParam(value = "pageSize", defaultValue = "5") Integer pageSize,
            @RequestParam(value = "orderBy", defaultValue = "teacherSelectCourseId") String orderBy,
            @RequestBody edu.hubu.commons.mybatis.model.TeacherSelectCourse teacherSelectCourse
    );

    @PostMapping("/teacherSelectCourse/")
    CommonResult save(@RequestBody TeacherSelectCourse teacherSelectCourse);

    @PutMapping("/teacherSelectCourse/{id}")
    CommonResult update(@PathVariable("id") String id, @RequestBody TeacherSelectCourse teacherSelectCourse);


    @DeleteMapping("/teacherSelectCourse/delete/{id}")
    CommonResult deleteById(@PathVariable("id") String id);
}

File: educational-system-commons/src/main/java/edu/hubu/commons/service/teacher/ITeacherSelectCourseService.kt
package edu.hubu.commons.service.teacher

import edu.hubu.commons.model.teacher.TeacherSelectCourse;

/**
* @author moonlan
* date 2021/1/25 下午2:07
*/
interface ITeacherSelectCourseService {
    fun save(t: TeacherSelectCourse): TeacherSelectCourse
    fun deleteById(id: String): Boolean
    fun updateById(id: String, t: TeacherSelectCourse): TeacherSelectCourse
    fun selectOneById(id: String): TeacherSelectCourse
    fun selectAllOrderBy(order: String = "teacherSelectCourseId"): List<TeacherSelectCourse>
    fun selectAllPageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "teacherSelectCourseId"): List<TeacherSelectCourse>
    fun selectLikePageOrderBy(currentPage: Int = 1, pageSize: Int = 5, order: String = "teacherSelectCourseId", t: edu.hubu.commons.mybatis.model.TeacherSelectCourse): List<edu.hubu.commons.mybatis.model.TeacherSelectCourse>
}
